microsoft/testsuites/firewalld/firewalldsuite.py

```diff
@@ -0,0 +1,229 @@
+        "/etc/firewalld/firewalld.conf", "^IPv6_rpfilter=", sudo=True
+    )
+
+    if "=yes" in ipv6_rpfilter_str:
+        kconfig = node.tools[KernelConfig]
+        fib_inet = kconfig.is_enabled("CONFIG_FIB_INET")
+        fib_ipv6 = kconfig.is_enabled("CONFIG_FIB_IPV6")
+        if not fib_inet and not fib_ipv6:
+            # kernel does not support the needed configs
+            return False
+
+    return True
+
+
+def _update_test_result(
+    node: Node,
+    log_file: str,
+    log_dir: str,
+    result: TestResult,
+) -> None:
+    # Update TestResult data to add:
+    # - summary of the tests (Total/Skipped/Failed)
+    # - list of failed & skipped tests
+    # - overall status (PASSED/FAILED)
+    cat = node.tools[Cat]
+    ls = node.tools[Ls]
+    if not ls.path_exists(log_file):
+        raise LisaException(
+            f"Consolidated firewalld test log path: {log_file} doesn't exist, "
+            "please check if testing runs well or not."
+        )
+
+    total_tests = None
+    failed_tests = None
+    skipped_tests = None
+    num_regex = r"\d+ "
+
+    total_str = cat.read_with_filter(f"{log_file}", "tests were run", sudo=True)
+    m = re.search(num_regex, total_str)
+    if m:
+        total_tests = m[0]
+
+    failed_str = cat.read_with_filter(f"{log_file}", "failed unexpectedly", sudo=True)
+    m = re.search(num_regex, failed_str)
+    if m:
+        failed_tests = m[0]
+
+    skipped_str = cat.read_with_filter(f"{log_file}", "were skipped", sudo=True)
+    m = re.search(num_regex, skipped_str)
+    if m:
+        skipped_tests = m[0]
+
+    # Logs dir contains the list of failed tests.
+    current_failures: List[str] = []
+    failed_tests_list = ls.list_dir(log_dir, sudo=True)
+    for test in failed_tests_list:
+        # each entry in this list looks like
+        # "/usr/share/firewalld/testsuite/testsuite.dir/123/"
+        # using / as delimiter, we extract the test case number
+        # in this case 123
+        current_failures.append(test.split("/")[-2])
+
+    # Note: There are known failures due to missing support for `FIB`
+    # based expressions in kernel.
+    # This is a constant list of failing test cases in each run.
+    known_failures = [
+        "061",
+        "107",
+        "120",
+        "124",
+        "192",
+        "193",
+        "194",
+        "195",
+        "196",
+        "197",
+        "200",
+        "240",
+        "252",
+        "306",
+        "324",
+    ]
+
+    # Aarch64 kernel config has some additional FIB enabled
+    # configs compared to x86_64.
+    # Aarch64 failures list is thus a subset of known failures.
+    if not set(current_failures).issubset(set(known_failures)):
+        raise LisaException(
+            "Found unexpected failures, check logs for details\n"
+            f"expected failure list: {known_failures}\n"
+            f"actual failure list:   {current_failures}\n"
+        )
+
+    # Update the test result data
+    result.set_status(
```

Comment:
Remove the requirement to set the test case status to "PASSED". It's not recommended to add error messages to passed test cases. If detailed subtest information is needed, send the subtest results instead.

Reply:
No @squirrelsc. I don't want to get subtest results here.